### PR TITLE
feat:Added PIN generation

### PR DESCRIPTION
**Pull request by Narasimha1997**
Added PIN generation function that generates 4 digits PIN based on UUID provided as per the algorithm mentioned in #150 .
Modified `CharLink.js` to return `pin` and `pinCreationTime` as per #152

### CODE REVIEW

File: backend/api/chatLink/utils/link.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 const uuidv4 = require('uuid').v4;
+const generatePin = require('./pin').generatePIN
```
Comment (muke1908) [inline backend/api/chatLink/utils/link.js:2]:
```suggestion
const { generatePin } = require('./pin')
```

File: backend/api/chatLink/utils/link.js [hunk 2]
```diff
@@ -9,7 +10,9 @@ const generateLink = () => {
     link: `/chat/${hash}`,
     absoluteLink: `${chatLinkDomain}/chat/${hash}`,
     expired: false,
-    deleted: false
+    deleted: false,
+    pin : generatePin(hash, pinLength = 4),
```
Comment (muke1908) [inline backend/api/chatLink/utils/link.js:14]: Although the chances of duplicate are extremely low I would suggest having a check against db before creating new record at https://github.com/muke1908/chat-e2ee/blob/1e92f018aad2b19e4257952d064ce44dd36f1e00/backend/api/chatLink/index.js#L29

**Review by muke1908 — CHANGES_REQUESTED** [review]

File: backend/api/chatLink/utils/link.js (hunk 2 above)
Comment (Narasimha1997) [reply backend/api/chatLink/utils/link.js:14]: Sure, I will add this in the next commit

Comment (Narasimha1997) [reply backend/api/chatLink/utils/link.js]: Added this functionality:

```javascript
    let link = generateLink();
    //This ensures, PINs won't clash each other
    //Base case loop is not even executed
    //worst case, loop can take 2 or more iterations
    while(findOneFromDB({pin : link.pin}, LINK_COLLECTION)) {
      link = generateLink()
    }
    await insertInDb(link, LINK_COLLECTION);
    return res.send(link);
```

**Pull request merged**